oap-server/microbench/src/main/java/org/apache/skywalking/oap/server/microbench/core/config/group/uri/RegexVSQuickMatchBenchmark.java

```diff
@@ -0,0 +1,166 @@
+            rule.addRule("service1", "/products/{var}");
+            rule.addRule("service1", "/products/{var}/detail");
+            rule.addRule("service1", "/sales/{var}/1");
+            rule.addRule("service1", "/sales/{var}/2");
+            rule.addRule("service1", "/sales/{var}/3");
+            rule.addRule("service1", "/sales/{var}/4");
+            rule.addRule("service1", "/sales/{var}/5");
+            rule.addRule("service1", "/sales/{var}/6");
+            rule.addRule("service1", "/sales/{var}/7");
+            rule.addRule("service1", "/sales/{var}/8");
+            rule.addRule("service1", "/sales/{var}/9");
+            rule.addRule("service1", "/sales/{var}/10");
+            rule.addRule("service1", "/sales/{var}/11");
+            rule.addRule("service1", "/employees/{var}/profile");
+        }
+
+        public StringFormatGroup.FormatResult match(String serviceName, String endpointName) {
+            return rule.format(serviceName, endpointName);
+        }
+    }
+
+    @Benchmark
+    public void matchFirstRegex(Blackhole bh, RegexVSQuickMatchBenchmark.RegexMatch formatClass) {
+        bh.consume(formatClass.match("service1", "/products/123"));
+    }
+
+    @Benchmark
+    public void matchFirstQuickUriGrouping(Blackhole bh, RegexVSQuickMatchBenchmark.QuickMatch formatClass) {
+        bh.consume(formatClass.match("service1", "/products/123"));
+    }
+
+    @Benchmark
+    public void notMatchRegex(Blackhole bh, RegexVSQuickMatchBenchmark.RegexMatch formatClass) {
+        bh.consume(formatClass.match("service1", "/employees/123"));
+    }
+
+    @Benchmark
+    public void notMatchQuickUriGrouping(Blackhole bh, RegexVSQuickMatchBenchmark.QuickMatch formatClass) {
+        bh.consume(formatClass.match("service1", "/employees/123"));
+    }
+}
+
+/**
+ * # JMH version: 1.25
+ * # VM version: JDK 11.0.18, OpenJDK 64-Bit Server VM, 11.0.18+10
+ * # VM invoker: /Users/wusheng/Library/Java/JavaVirtualMachines/temurin-11.0.18/Contents/Home/bin/java
+ * # VM options: -ea --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED -Didea.test.cyclic.buffer.size=1048576 -javaagent:/Applications/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=53714:/Applications/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8 -Xmx512m -Xms512m -XX:MaxDirectMemorySize=512m -XX:BiasedLockingStartupDelay=0 -Djmh.executor=CUSTOM -Djmh.executor.class=org.apache.skywalking.oap.server.microbench.base.AbstractMicrobenchmark$JmhThreadExecutor
+ * # Warmup: 1 iterations, 10 s each
+ * # Measurement: 1 iterations, 10 s each
+ * # Timeout: 10 min per iteration
+ * # Threads: 4 threads, will synchronize iterations
+ * # Benchmark mode: Throughput, ops/time
+ * # Benchmark: org.apache.skywalking.oap.server.microbench.core.config.group.uri.RegexVSQuickMatchBenchmark.matchFirstQuickUriGrouping
+ *
+ Benchmark                                                                            Mode  Cnt         Score   Error   Units
```

Comment:
@Superskyyy This is the temp benchmark for new kinds of grouping.

@wankai123 You had known this, do you think it is worth we replace the regex permanently? From a code perspective, it is fine to keep both.
For now, this new mode doesn't work from the config file or/and dynamic configuration. 
It only works for AI/ML responses.